src/util/fingerprint.cpp

```diff
@@ -1,2 +1,61 @@
 #include "util/fingerprint.hpp"
-#include "util/fingerprint_impl.hpp"
+#include "util/exception.hpp"
+#include "util/exception_utils.hpp"
+#include "util/version.hpp"
+
+#include <boost/uuid/name_generator.hpp>
+#include <boost/uuid/uuid_generators.hpp>
+
+#include <cstring>
+
+#include <algorithm>
+#include <string>
+
+namespace osrm
+{
+namespace util
+{
+
+FingerPrint FingerPrint::GetValid()
+{
+    FingerPrint fingerprint;
+
+    // 4 chars, 'O','S','R','N' - note the N instead of M, v1 of the fingerprint
+    // used M, so we add one and use N to indicate the newer fingerprint magic number.
+    // Bump this value if the fingerprint format ever changes.
+    fingerprint.magic_number = {{'O', 'S', 'R', 'N'}};
+    // fingerprint.magic_number[1] = 'S';
+    // fingerprint.magic_number[2] = 'R';
+    // fingerprint.magic_number[3] = 'N';
+    fingerprint.major_version = static_cast<std::uint8_t>(atoi(OSRM_VERSION_MAJOR));
+    fingerprint.minor_version = static_cast<std::uint8_t>(atoi(OSRM_VERSION_MINOR));
+    fingerprint.patch_version = static_cast<std::uint8_t>(atoi(OSRM_VERSION_PATCH));
+    fingerprint.unused = 0x00;
+
+    return fingerprint;
+}
+
+int FingerPrint::GetMajorVersion() const { return major_version; }
+int FingerPrint::GetMinorVersion() const { return minor_version; }
+int FingerPrint::GetPatchVersion() const { return patch_version; }
+
+bool FingerPrint::IsMagicNumberSame(const FingerPrint &other) const
```

Comment:
I think better to have member operators, like `operator==`, `operator<` and `operator<<` to hide details about magic and versioning.  It will be easy later modify internals to have e.g.  non-numeric release version. 
`operator<` will allow version range checks, although at the moment there is no use case for it.